app/services/power_manager/virt.rb

```diff
@@ -20,31 +20,23 @@ def state
       vm.state
     end
 
-    (SUPPORTED_ACTIONS - ['state']).each do |method|
-      define_method method do
-        vm.send(action_map[method.to_sym])
-      end
+    def default_action(action)
+      vm.send(action)
+    end
+
+    def action_map
+      super.deep_merge({
+                         :on       => 'start',
+                         :off      => 'stop',
+                         :soft     => 'reboot',
+                         :cycle    => 'reset',
+                         :status   => 'state',
+                         :state    => {:action => :state}
+                       })
     end
 
     private
 
     attr_reader :vm
-
-    def action_map
-      {
-        :on       => 'start',
-        :off      => 'stop',
-        :soft     => 'reboot',
-        :reboot   => 'reboot',
-        :cycle    => 'reset',
-        :status   => 'state',
-        :start    => 'start',
-        :stop     => 'stop',
```

Comment:
Removing these (poweroff, reboot, state...) breaks the API somewhat doesn't it? For instance, I'd expect to be able to send 'poweroff' to all my hosts and power off both physical and virtual hosts automatically.